packages/odyssey-react-mui/src/Radio.tsx

```diff
@@ -95,10 +96,24 @@ const Radio = ({
   );
 
   const onChange = useCallback<NonNullable<MuiRadioProps["onChange"]>>(
-    (event, checked) => onChangeProp?.(event, checked),
-    [onChangeProp],
+    (event, checked) => {
+      if (isReadOnly) {
+        event.preventDefault();
+        return;
+      }
+      onChangeProp?.(event, checked);
+    },
```

Comment:
Instead of early-fail, I wonder if we can do `if-else` instead:
```suggestion
    (event, checked) => {
      if (isReadOnly) {
        event.preventDefault();
      }
      else {
        onChangeProp?.(event, checked);
      }
    },
```

That might be more clear. The `return` in that `if` statement can disappear if you're not looking for it. I've used a lot of early-fail, but I've avoided in more recent years to make it clear we're doing one or the other, and `return` is always at the bottom.